util/suffix.c

```diff
@@ -65,36 +68,87 @@ const char *suffix_si_get(double *value)
 	return suffix;
 }
 
-uint64_t suffix_si_parse(const char *value, bool *suffixed)
+int suffix_si_parse(const char *str, char **endptr, uint64_t *val)
 {
-	char *suffix;
-	double ret;
-	struct si_suffix *s;
-
-	errno = 0;
-	ret = strtod(value, &suffix);
-	if (errno)
+	unsigned long long num, frac;
+	char *sep, *tmp;
+	int frac_len, len, i;
+
+	num = strtoull(str, endptr, 0);
+	if (str == *endptr ||
+	    ((num == ULLONG_MAX) && errno == ERANGE))
+		return -EINVAL;
+
+	/* simple number, no decimal point not suffix */
+	if (*endptr[0] == '\0') {
```

Comment:
This also should be fixed as same with other casting `endptr` parts. (By the way it seems that it is same with the case without the casting for the pointer of pointer as `(*endptr)[n]` and `*endptr[n]` are same but not sure..)
```diff
- *endptr[0]
+ (*endptr)[0]
```

Reply:
Oh I missed those. I'll fix them as well.

The operator precedence says  `[]` wins over `*`. That means `*endptr[1]` will point to invalid memory. It happens 
that `*endptr[0]` and `(*endptr)[0]` point to the same memory address.

Reply:
Thanks for your reply and explanations. Yes I know the precedence but now I could underand the diffence between `*endptr[1]` and `(*endptr)[1]` but same for `*endptr[0]` and `(*endptr)[0]`.
P.S. Sorry my same comments were repeated as missed so deleted them. (Sometimes github comments seems not updated to add so repeated to push the comment button.)